data/templates/metronome/domain.tpl.cfg.lua

```diff
@@ -1,15 +1,96 @@
   }
   authentication = "ldap2"
   ldap = {
-     hostname      = "localhost",
-     user = {
-       basedn        = "ou=users,dc=yunohost,dc=org",
-       filter        = "(&(objectClass=posixAccount)(mail=*@{{ domain }})(permission=cn=xmpp.main,ou=permission,dc=yunohost,dc=org))",
-       usernamefield = "mail",
-       namefield     = "cn",
-       },
+    hostname = "localhost",
+    user = {
+      basedn = "ou=users,dc=yunohost,dc=org",
+      filter = "(&(objectClass=posixAccount)(mail=*@{{ domain }})(permission=cn=xmpp.main,ou=permission,dc=yunohost,dc=org))",
+      usernamefield = "mail",
+      namefield = "cn",
+    },
   }
+
+
+-- Discovery items
+disco_items = {
+  { "muc.{{ domain }}" },
+  { "pubsub.{{ domain }}" },
+  { "jabber.{{ domain }}" },
+  { "vjud.{{ domain }}" }
+};
+
+--contact_info = {
+--  abuse = { "mailto:abuse@{{ domain }}", "xmpp:admin@{{ domain }}" };
+--  admin = { "mailto:root@{{ domain }}", "xmpp:admin@{{ domain }}" };
+--};
+
+-- BOSH configuration (mod_bosh)
+consider_bosh_secure = true
+cross_domain_bosh = true
+
+-- WebSocket configuration (mod_websocket)
+consider_websocket_secure = true
+cross_domain_websocket = true
+
+-- Disable account creation by default, for security
+allow_registration = false
+
+-- Use LDAP storage backend for all stores
+storage = "ldap"
+
+-- Logging configuration
+log = {
+  info = "/var/log/metronome/metronome.log"; -- Change 'info' to 'debug' for verbose logging
+  error = "/var/log/metronome/metronome.err";
+  -- "*syslog"; -- Uncomment this for logging to syslog
+  -- "*console"; -- Log to the console, useful for debugging with daemonize=false
+}
+
+------ Components ------
+-- You can specify components to add hosts that provide special services,
+-- like multi-user conferences, and transports.
+
+---Set up a local BOSH service
+Component "localhost" "http"
+  modules_enabled = { "bosh" }
+
+---Set up a MUC (multi-user chat) room server
+Component "muc.{{ domain }}" "muc"
+  name = "{{ domain }} Chatrooms"
+
+  modules_enabled = {
+    "muc_limits";
+    "muc_log";
+    "muc_log_mam";
+    "muc_log_http";
+    "muc_vcard";
+  }
+
+  muc_event_rate = 0.5
+  muc_burst_factor = 10
+
+---Set up a PubSub server
+Component "pubsub.{{ domain }}" "pubsub"
+  name = "{{ domain }} Publish/Subscribe"
+
+  unrestricted_node_creation = true -- Anyone can create a PubSub node (from any server)
+
+---Set up a HTTP Upload service
+Component "jabber.{{ domain }}" "http_upload"
+  name = "{{ domain }} Sharing Service"
+
+  http_file_size_limit = 6*1024*1024
+  http_file_quota = 60*1024*1024
+  http_upload_file_size_limit = 100 * 1024 * 1024 -- bytes
+  http_upload_quota = 10 * 1024 * 1024 * 1024 -- bytes
+  http_upload_path = "/var/lib/metronome/jabber.{{ domain }}"
+  http_external_url = "https://jabber.{{ domain }}/"
+
+---Set up a VJUD service
+Component "vjud.{{ domain }}" "vjud"
+  ud_disco_name = "{{ domain }} User Directory"
```

Comment:
You are moving global configuration to the domain specific configuration template. There will be duplicate configurations due to multiple domains configurations. Why did you do that?

Reply:
There won't be any duplicate since each component name contains the {{ domain }} variable.
Or did I miss something?

Just to be clear, this change (moving global configuration to the virtualhost config) is not related to the main goal of that PR (make http upload working). But it's a prerequisite.
I consider this is a long-standing bug in yunohost: only the main yunohost domain can host a full-featured XMPP server, because there are no components defined for the other domains.

Reply:
We would get a `muc` (multi-user-channel) `pubsub`, `xmpp-upload`, and `vjud` per domain? Why for? Does anyone would see a benefit in full-featured XMPP on all of their domains? Would that open new potential flaws?

---

Why not creating two PRs? One to fix this `http-upload` and the other to add components to all domains? Wouldn’t that be easier to review? The title and the description of this PR is about fixing this `http-upload` not adding components to all domains. This is very confusing. Let’s make things KISS when possible. We do not need this complexity here.

Reply:
I too was thinking about making two PR. It will be a bit more work for me but I it'll make things easier for reviewers.

That said, I still consider that each domain should have its own xmpp components. Otherwise users of domainB and domainC would have to use chatrooms of domainA for example. That would make Yunohost unsuitable for hosters like [CHATONS](https://chatons.org/en) who would like to offer full-featured XMPP services for multiple different groups of persons.